src/routes/users.rs

```diff
@@ -54,6 +92,160 @@ async fn post_login(
     Ok(Redirect::to(&redirect_to).into_response())
 }
 
+async fn get_login_google(
+    State(state): State<AppState>,
+    session: Session,
+) -> ResponseResult<Response> {
+
+    let google_client_id = ClientId::new(
```

Comment:
There is lots of code that deals with the set up of the `CoreClient` struct here. Can we do this when the application starts up? That way, if there is an error in the OAuth logic, we will see it earlier. It will also allow us to deduplicate lots of similar code in both oauth handlers.

I'm imagining something like this:
```rs
pub struct AppState {
    // Use this field instead of `oauth_google_client_id` and `oauth_google_client_secret`
    pub oauth_client: CoreClient
}
```

Reply:
Got it, I'll refactor the code again later, probs after implementing #36